plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/DependencyGroup.kt

```diff
@@ -0,0 +1,172 @@
+package de.fayard.refreshVersions.core
+
+import de.fayard.refreshVersions.core.internal.ArtifactVersionKeyRule
+import de.fayard.refreshVersions.core.internal.InternalRefreshVersionsApi
+import org.gradle.kotlin.dsl.IsNotADependency
+
+abstract class DependencyGroup(
+    group: String,
+    rawRule: String? = null,
+    usePlatformConstraints: Boolean = false
+) : IsNotADependency, AbstractDependencyGroup(
+    group = group,
+    rawRule = rawRule,
+    usePlatformConstraints = usePlatformConstraints
+)
+
+@OptIn(PrivateForImplementation::class)
+abstract class DependencyNotationAndGroup(
+    group: String,
+    name: String,
+    rawRule: String? = null,
+    usePlatformConstraints: Boolean = false
+) : AbstractDependencyGroup(
+    group = group,
+    rawRule = rawRule,
+    usePlatformConstraints = usePlatformConstraints
+), DependencyNotation by object : DependencyNotationImpl(group = group, name = name) {
+    override fun shouldUsePlatformConstraints(): Boolean = false
+} {
+    val artifactPrefix: String get() = withoutVersion()
+}
+
+@RequiresOptIn
+private annotation class PrivateForImplementation
+
+interface DependencyNotation : CharSequence {
+    fun withVersionPlaceholder(): String
+    fun withVersion(version: String): String
+    fun withoutVersion(): String
+    operator fun invoke(version: String?): String
+
+    override fun toString(): String
+
+    @PrivateForImplementation
+    val externalImplementationGuard: Nothing
+}
+
+@OptIn(PrivateForImplementation::class)
+private abstract class DependencyNotationImpl(
+    group: String,
+    name: String
+) : DependencyNotation {
+
+    protected abstract fun shouldUsePlatformConstraints(): Boolean
+
+    override fun withVersionPlaceholder(): String = this(version = "_")
+    override fun withVersion(version: String): String = this(version = version)
+    override fun withoutVersion(): String = this(version = null)
+
+    override operator fun invoke(version: String?): String = when (version) {
+        null -> artifactPrefix
```

Comment:
We can use `.isNullOrBlank()` instead, for the least surprising behavior in case `""` or `"   "` are passed.